trace-mini-agent/src/stats_flusher.rs

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use async_trait::async_trait;
+use log::{debug, error, info};
+use std::{sync::Arc, time};
+use tokio::sync::{mpsc::Receiver, Mutex};
+
+use datadog_trace_protobuf::pb;
+use datadog_trace_utils::stats_utils;
+
+#[async_trait]
+pub trait StatsFlusher {
+    /// Starts a stats flusher that listens for stats payloads sent to the tokio mpsc Receiver,
+    /// implementing flushing logic that calls flush_stats.
+    async fn start_stats_flusher(&self, mut rx: Receiver<pb::ClientStatsPayload>);
+    /// Flushes stats to the Datadog trace stats intake.
+    async fn flush_stats(&self, traces: Vec<pb::ClientStatsPayload>);
+}
+
+#[derive(Clone)]
+pub struct ServerlessStatsFlusher {}
+
+#[async_trait]
+impl StatsFlusher for ServerlessStatsFlusher {
+    async fn start_stats_flusher(&self, mut rx: Receiver<pb::ClientStatsPayload>) {
+        let buffer: Arc<Mutex<Vec<pb::ClientStatsPayload>>> = Arc::new(Mutex::new(Vec::new()));
+
+        let buffer_producer = buffer.clone();
+        let buffer_consumer = buffer.clone();
+
+        tokio::spawn(async move {
+            while let Some(stats_payload) = rx.recv().await {
+                let mut buffer = buffer_producer.lock().await;
+                buffer.push(stats_payload);
+            }
+        });
+
+        loop {
+            tokio::time::sleep(time::Duration::from_millis(3000)).await;
```

Comment:
should we move that to a constant on top of the file? that might be easier to change if we want

Reply:
good idea, done